File: Anya_Seth/2-Pointer/Reverse Words in a String III.java
class Solution {
    public String reverseWords(String s) {
       char[] arr = s.toCharArray();
       int first = 0;
       int last = 0;
       int n =s.length();
       while(last < n){
        while(last < n && arr[last] != ' '){
            last++;
        }
            reverse(arr, first, last - 1);
            last++;
            first = last;
        
       }
       return new String(arr);
    }

    public void reverse(char[] arr, int first, int last){
        while(first < last){
            char temp = arr[last];
            arr[last] = arr[first];
            arr[first] = temp;
            first++;
            last--;
        }
    }
}

File: Lakshay Garg/Leetcode/Two_Pointers_Leetcode/Reverse_Words_in_a_String_III.cpp
class Solution {
public:
    string reverseWords(string s) {
        int i=0,j=0,n=s.size();
        while(j<n){
            if(s[j]==' '){
                reverse(s.begin()+i,s.begin()+j);
                i=j+1;
                j++;
            }
            else j++;
        }
        reverse(s.begin()+i,s.begin()+j);
        return s;
    }
};
